front/src/applications/operationalStudies/hooks/useTrackSections.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useCallback } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { InfraObjectWithGeometry } from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+export default function useCachedTrackSections(infraId: number, dispatch: AppDispatch) {
```

Comment:
Agreed for appDispatch, but clara told me we want to get the infra id out of the store and into the context. And while this is just a preliminary step I would still prefer to avoid adding another reliance to the store for the infra id.

Reply:
Ok seems legit :)